client/state/data-layer/wpcom-http/test/index.js

```diff
@@ -77,3 +80,99 @@ describe( '#queueRequest', () => {
 		}, 10 );
 	} );
 } );
+
+describe( '#fetcherMap', () => {
+	const wpcomReq = {};
+	const noopCallback = () => {};
+
+	describe( 'GET', () => {
+		let getFooAction;
+
+		beforeEach( () => {
+			wpcomReq.get = spy();
+			getFooAction = {
+				method: 'GET',
+				path: '/foo',
+			};
+		} );
+
+		it( 'should send with query', () => {
+			getFooAction.query = {
+				apiVersion: 'v2.0',
+				apiNamespace: 'wp/v2',
+			};
+
+			fetcherMap( 'GET', wpcomReq )( getFooAction, noopCallback );
+
+			expect( wpcomReq.get ).to.have.been.calledWith(
+				{ path: '/foo' },
+				{
+					apiVersion: 'v2.0',
+					apiNamespace: 'wp/v2',
+				},
+				noopCallback
+			);
+		} );
+	} );
+
+	describe( 'POST', () => {
+		let postFooAction;
+
+		beforeEach( () => {
+			wpcomReq.post = spy();
+			postFooAction = {
+				method: 'POST',
+				path: '/foo',
+			};
+		} );
+
+		it( 'should send formData with query', () => {
+			postFooAction.formData = [ [ 'foo', 'bar' ], ];
+			postFooAction.query = { apiVersion: 'v1.1' };
```

Comment:
Yep, I think I got confused with the default value used for `apiVersion` in the `http` action creator: https://github.com/Automattic/wp-calypso/blob/master/client/state/data-layer/wpcom-http/actions.js#L23

Now I wonder if there is a special case for `v1` or if we should also remove the `v` in there? In #13022 @dmsnell got rid of the default but we're still using `v1` in the tests, might be worth tackling it there as well.